### DISCUSSION TITLE
Access previous exports?

### DISCUSSION THREAD

**Discussion by ajstorrs** [ReagentX/imessage-exporter#397] [Q&A] [answered]
Hi - apologies in advance for the potentially basic question: I performed an export a few months ago and am attempting to re-export, but some of the conversations are missing a significant amount of data compared with the previous export. The Mac I am exporting from has been turned off and has not had internet connection since the last export. Any thoughts? Can I somehow recreate the previous export? Thank you!

**Comment by ReagentX** [comment]
What version of `imessage-exporter` was the previous export made using? Are the messages you expect to see still present in Messages.app?

**Reply by ajstorrs** [reply]
I'm not entirely certain, but pretty sure it was version 2.0.1. The messages are no longer in messages.app

**Reply by ReagentX** [reply]
Given they don't show up in the messages app anymore, I doubt `imessage-exporter` will be able to export them, because they're probably no longer present in the database.

If you try and install that version of `imessage-exporter`, do the messages show up in the exports? Were these orphaned or recovered messages, or just normal messages?

**Reply by ajstorrs** [reply]
Unfortuantely, they still do not appear. They were normal messages.  

Are you aware of any way to recover, either with your tool or other software?

Thank you for creating this tool - it's truly amazing, much appreciated!

**Reply by ReagentX** [reply, marked as answer]
In that case, it is very likely that the relevant message rows are no longer present in the database on the machine you are trying to export from. 

As for recovery, `imessage-exporter` automatically performs several [recovery steps](https://github.com/ReagentX/imessage-exporter/blob/develop/docs/faq.md#are-messages-deleted-from-the-messages-app-erased-from-the-database). Any messages that cannot be linked to a specific chat are saved to the orphaned messages file. Beyond that, there isn't much you can do if the source data is gone.

This is why `imessage-exporter` will not let you overwrite already-exported files–by design, `imessage-exporter` should ensure integrity of exported data.
